extension/apmconfigextension/opamp_callbacks.go

```diff
@@ -45,11 +58,22 @@ type agentInfo struct {
 	lastConfigHash        apmconfig.LastConfigHash
 }
 
-func newRemoteConfigCallbacks(configClient apmconfig.RemoteConfigClient, logger *zap.Logger) *remoteConfigCallbacks {
+func newRemoteConfigCallbacks(configClient apmconfig.RemoteConfigClient, ttlConfig cacheConfig, logger *zap.Logger) (*remoteConfigCallbacks, error) {
+	cache, err := freelru.NewSharded[string, *agentInfo](ttlConfig.capacity, func(key string) uint32 {
+		h := fnv.New32a()
```

Comment:
FNV1a is one of the worst algorithms for hashing in regards to performance and randomness. Though it doesn't matter if the hash/lru operations are part of a hot path.